File: app/src/main/java/com/electrosim/ViewThread.java
package com.electrosim;

import android.graphics.Canvas;
import android.view.SurfaceHolder;

public class ViewThread extends Thread {
    private Container mPanel;
    private SurfaceHolder mHolder;
    private boolean mRun = false;
    
    public ViewThread(Container panel) {
        mPanel = panel;
        mHolder = mPanel.getHolder();
    }
    
    public void setRunning(boolean run) {
        mRun = run;
    }
    
    @Override
    public void run() {
        Canvas canvas = null;
       while (mRun) {
            canvas = mHolder.lockCanvas();
            if (canvas != null) {
            	//canvas.scale(2, 2);
                mPanel.doDraw(canvas);
                mHolder.unlockCanvasAndPost(canvas);
            }
        }
    }
}

File: app/src/main/java/com/electrosim/ViewThread.kt
package com.electrosim

import android.graphics.Canvas
import android.view.SurfaceHolder

class ViewThread(private val mPanel: Container) : Thread() {
    private val mHolder: SurfaceHolder = mPanel.holder
    private var mRun = false
    
    fun setRunning(run: Boolean) {
        mRun = run
    }
    
    override fun run() {
        var canvas: Canvas?
        while (mRun) {
            canvas = mHolder.lockCanvas()
            if (canvas != null) {
                mPanel.doDraw(canvas)
                mHolder.unlockCanvasAndPost(canvas)
            }
        }
    }
} 